### PR TITLE
Feature/issue23

### PR DESCRIPTION
**Pull request by polyglotpiglet**
changes to server, client, integration tests for issue 23 (activity verification for preparation of prepared statements)

Will look at updating docs.

### CODE REVIEW

File: java-client/src/test/java/org/scassandra/http/client/ActivityClientTest.java [hunk 1]
```diff
@@ -224,8 +225,68 @@ public void testDeletingOfPreparedExecutionHistoryFailure() {
...
+        //then
+        verify(deleteRequestedFor(urlEqualTo(preparedStatementPreparationUrl)));
+    }
+
+    @Test(expected = ActivityRequestFailed.class)
+    public void testDeletingOfPreparationsHistoryFailure() {
+        //given
+        stubFor(delete(urlEqualTo(preparedStatementPreparationUrl))
+                .willReturn(aResponse().withStatus(200).withFault(Fault.RANDOM_DATA_THEN_CLOSE)));
+        //when
```
Comment (chbatey) [inline java-client/src/test/java/org/scassandra/http/client/ActivityClientTest.java]: I love wiremocks RANDOM_DATA_THEN_CLOSE ll

File: server/src/main/scala/org/scassandra/server/priming/ActivityLog.scala [hunk 2]
```diff
@@ -21,36 +21,54 @@ import org.scassandra.server.cqlmessages.types.ColumnType
...
 class ActivityLog extends LazyLogging {
 
-
-  var connections: List[Connection] = List()
-  var queries: List[Query] = List()
-  var preparedStatementExecutions: List[PreparedStatementExecution] = List()
+  var connections : List[Connection] = List()
+  var queries : List[Query] = List()
+  var preparedStatementPreparations : List[PreparedStatementPreparation] = List()
+  var preparedStatementExecutions : List[PreparedStatementExecution] = List()
```
Comment (chbatey) [inline server/src/main/scala/org/scassandra/server/priming/ActivityLog.scala]: delete this space :)

File: server/src/test/scala/org/scassandra/server/e2e/PreparedStatementPreparationVerificationTest.scala [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import org.scalatest.concurrent.ScalaFutures
+import org.scassandra.server.AbstractIntegrationTest
+import org.scassandra.server.priming.{PrimingJsonImplicits, PreparedStatementPreparation}
+import spray.json._
+
+class PreparedStatementPreparationVerificationTest extends AbstractIntegrationTest with ScalaFutures {
+
+  import PrimingJsonImplicits._
+
+  val primePreparedSinglePath = "http://localhost:8043/prepared-statement-preparation"
```
Comment (chbatey) [inline server/src/test/scala/org/scassandra/server/e2e/PreparedStatementPreparationVerificationTest.scala]: I need to remove this hard coding of host/port but don't worry about it we'll raise a separate issue to get rid of the duplication for all e2e tests in the server

File: java-client/src/main/java/org/scassandra/http/client/PreparedStatementPreparation.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
+package org.scassandra.http.client;
+
+public final class PreparedStatementPreparation {
```
Comment (chbatey) [inline java-client/src/main/java/org/scassandra/http/client/PreparedStatementPreparation.java]: Could you add an EqualsVerifier test? Like for BatchExecution as people will reply on the equals method in their tests.

**Comment by chbatey** [comment]
Looks great - a few very minor comments.

**Comment by tolbertam** [comment]
Looks good to me! :+1:

File: server/src/main/scala/org/scassandra/server/priming/routes/ActivityVerificationRoute.scala [hunk 5]
```diff
@@ -56,21 +57,36 @@ trait ActivityVerificationRoute extends HttpService with LazyLogging {
           }
         }
     } ~
-    path("prepared-statement-execution") {
+    path("prepared-statement-preparation") {
       get {
         complete {
-          logger.debug("Request for recorded prepared statement executions")
-          activityLog.retrievePreparedStatementExecutions()
+          logger.debug("Request for record prepared statement preparations")
```
Comment (chbatey) [inline server/src/main/scala/org/scassandra/server/priming/routes/ActivityVerificationRoute.scala]: recorded :)

File: server/src/test/scala/org/scassandra/server/priming/routes/ActivityVerificationRouteTest.scala [hunk 6]
```diff
@@ -97,6 +98,39 @@ class ActivityVerificationRouteTest extends FunSpec with BeforeAndAfter with Mat
...
+    it("Should return prepared statement preparations from ActivityLog - no activity") {
+      activityLog.clearPreparedStatementPreparations()
+
+      Get("/prepared-statement-preparation") ~> activityVerificationRoute ~> check {
+        val response = responseAs[List[PreparedStatementPreparation]]
+        response.size should equal(0)
+      }
+    }
+
+    it("Should return queries from ActivityLog - single query") {
```
Comment (chbatey) [inline server/src/test/scala/org/scassandra/server/priming/routes/ActivityVerificationRouteTest.scala]: test name: not for a single query, for a single prepare

File: server/src/test/scala/org/scassandra/server/priming/routes/ActivityVerificationRouteTest.scala [hunk 7]
```diff
@@ -97,6 +98,39 @@ class ActivityVerificationRouteTest extends FunSpec with BeforeAndAfter with Mat
...
+
+      Get("/prepared-statement-preparation") ~> activityVerificationRoute ~> check {
+        val response = responseAs[List[PreparedStatementPreparation]]
+
+        response.size should equal(1)
+        response.head.preparedStatementText should equal(preparedStatementText)
+      }
+    }
+
+    it("Should clear queries for a delete") {
```
Comment (chbatey) [inline server/src/test/scala/org/scassandra/server/priming/routes/ActivityVerificationRouteTest.scala]: test name: delete prepared preparations not queries

**Comment by chbatey** [comment]
Thanks for the updates, I added a few more comments but I'll do these on commit.

**Comment by chbatey** [comment]
@polyglotpiglet I am doing  a released with your changes and all the others in master. I am going to hold off releasing the docs until you add them for this feature.

**Pull request merged**